core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -74,7 +74,9 @@
     private Duration requiredWorkersMaxWait = new Duration(5, TimeUnit.MINUTES);
 
     private RetryPolicy retryPolicy = RetryPolicy.NONE;
-    private int retryAttempts = 4;
+    private int queryRetryAttempts = 4;
+    private int taskRetryAttemptsPerTask = 2;
+    private int taskRetryAttemptsOverall = Integer.MAX_VALUE;
```

Comment:
I am not super attached to it. And the default value is `Integer.MAX_VALUE` so by default it is ineffective. I think we can rid of it.
Then we should rename `taskRetryAttemptsPerTask` to `taskRetryAttempts`. @arhimondr makes sense?

Reply:
This property can be used as a safeguard from a retry avalanche, when all the tasks suddenly start failing? Though this is more of a hypothetical thinking, not sure how useful is it going to be in practice. I don't really have a strong opinion here. Since it's already there and implemented I'm fine with keeping it. But if you feel like it won't be particularly useful you should also feel free to remove it.